core/src/main/scala/org/apache/spark/deploy/master/Master.scala

```diff
@@ -658,19 +658,22 @@ private[deploy] class Master(
   private def startExecutorsOnWorkers(): Unit = {
     // Right now this is a very simple FIFO scheduler. We keep trying to fit in the first app
     // in the queue, then the second app, etc.
-    for (app <- waitingApps if app.coresLeft > 0) {
+    for (app <- waitingApps) {
```

Comment:
Is not be better to compare app.coresLeft whih coresPerExecutor? If the coresLeft less than coresPerExecutor, it will return directly